### DISCUSSION TITLE
hostRules credentials not correctly applied for bitbucket cloud submodule

### DISCUSSION THREAD

**Discussion by flichtenheld** [renovatebot/renovate#23299] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.6.1

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm running renovate against a repository on Bitbucket Cloud that has submodules, some also hosted on Bitbucket Cloud.
Checking out the top-level repository with `RENOVATE_USERNAME` and `RENOVATE_PASSWORD` works fine. I added a hostRules entry with the same username and password to be able to check out the submodules:

    hostRules": [
	{
            "matchHost": "bitbucket.org",
            "username": "frank_lichtenheld",
            "password": "PASSWORD"
	}
     ]

The rule matches, but then somehow the credentials provided are wrong, see log provided below.

It does seem to do something. If I remove the hostRules from the config, git instead prompts me for the password and I can provide it (but that obviously is not really suitable for something I want to run automated). So it seems to match the rule and provide *some* credentials. But in some way it is wrong. If I just run `git ls-remote --symref https://frank_lichtenheld:PASSWORD@bitbucket.org/WORKSPACE/REPO.git` manually with the same credentials it works fine.

Can I somehow get the unredacted error? I.e. find out what exactly it tries to use?

I saw there were various discussions and bugs related to authenticating to private submodules. But AFAICT none of these match exactly my problem.

Public reproduction of the problem is not possible, at least I don't know how one would do that. Which is why I would be happy to dig deeper when someone can give me a hint on how to do that.

Currently I run renovate in docker, like so:

    docker run -it -e RENOVATE_CONFIG_FILE=/usr/src/app/config.json -e RENOVATE_USERNAME=frank_lichtenheld -e RENOVATE_PASSWORD=PASSWORD -e GITHUB_COM_TOKEN=TOKEN -e LOG_LEVEL=debug --rm -v "$PWD/renovate/renovate-config.json:/usr/src/app/config.json" renovate/renovate:36

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Found hostRules username and password for url https://**redacted**@bitbucket.org/WORKSPACE/REPO.git (repository=WORKSPACE/PARENT_REPO)
    WARN: Error mapping git submodules during extraction (repository=WORKSPACE/PARENT_REPO, packageFile=.gitmodules)
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "--symref",
             "https://**redacted**@bitbucket.org/WORKSPACE/REPO.git",
             "HEAD"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "remote: Invalid credentials\nfatal: Authentication failed for 'https://bitbucket.org/WORKSPACE/REPO.git/'\n",
         "stack": "Error: remote: Invalid credentials\nfatal: Authentication failed for 'https://bitbucket.org/WORKSPACE/REPO.git/'\n\n    at Object.action (/opt/containerbase/tools/renovate/36.6.1/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/opt/containerbase/tools/renovate/36.6.1/node_modules/simple-git/src/lib/plugins/plugin-store.ts:29:29)\n    at /opt/containerbase/tools/renovate/36.6.1/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:127:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/opt/containerbase/tools/renovate/36.6.1/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:14)\n    at GitExecutorChain.<anonymous> (/opt/containerbase/tools/renovate/36.6.1/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:100:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/opt/containerbase/tools/renovate/36.6.1/node_modules/simple-git/dist/cjs/index.js:55:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
       }
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by maebli** [comment]
Have you found a solution? Seems to be a possible duplicate of my issue 
https://github.com/renovatebot/renovate/discussions/23812

Maybe upvote my issue to get traction on this

**Reply by flichtenheld** [reply]
No, made no progress so far. Even the debug log sounds like it should work but then simply doesn't. And since all the credentials are redacted in the log it is difficult to say what it actually does. The next step would probably to dive into the code and actually try to debug it. But since I lack any experience with the language and the code-base, I haven't dared to, yet.

**Comment by frjonsen** [comment]
I have been struggling with this for a few hours as well. Is this still broken, or am I just missing something?

**Reply by flichtenheld** [reply]
I never got it working and just gave up on it at some point. Never dug deep enough to understand the actual problem, either.

**Reply by aaijd** [reply]
https://github.com/renovatebot/renovate/discussions/36378
